### PR TITLE
Translations

### PR DESCRIPTION
**Pull request by davidheyman**
Locale support and translations

### CODE REVIEW

File: src/pages/index.js [hunk 1]
```diff
@@ -92,7 +99,7 @@ export async function getServerSideProps({ req }) {
     data: {
       post_stream: { posts },
     },
-  } = await axios.get(`${process.env.NEXT_PUBLIC_PAGE_URL}en-narratives-about/96.json`, {
+  } = await axios.get(`${process.env.NEXT_PUBLIC_PAGE_URL}${locale}-narratives-about/96.json`, {
```
Comment (buccalon) [inline src/pages/index.js:102]: Portuguese version has a different url: https://forum.imaginerio.org/t/pt-narratives-about/46

**Comment by davidheyman** [comment]
@buccalon This is ready for a full review. I'm sure you've seen the translation file here (src/hooks/useLocale.js) which will need fixing by a native speaker. Could you also check to make sure that none of the links have broken? I still need to add a link to switch between the languages but I'll figure out where that goes later.

**Review by martimpassos — CHANGES_REQUESTED** [review]

File: src/hooks/useLocale.js [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    size: 'Tamanho',
+    fullscreen: 'Tela cheia',
+    medium: 'Médio',
+    small: 'Pequeno',
+    externalMedia: 'Mídia externa',
+    features: 'Recursos',
+    newAnnotation: 'Nova anotação',
+    year: 'Ano',
+    layers: 'Camadas',
+    visuals: 'Mapas / Planos / Aerias',
```
[the remark this answers is not in the document]
Comment (buccalon) [reply src/hooks/useLocale.js:111]: Não deveria ser `Mapas / Projetos / Imagens Aéreas`?

[the remark this answers is not in the document]
Comment (martimpassos) [reply src/hooks/useLocale.js]: Pode ser, eu tentei diferenciar só porque temos imagens aéreas em `Vistas` também. Em `Aerials` são só levantamentos aerofotogramétricos ou imagens de satélite...

**Comment by buccalon** [comment]
@davidheyman Thanks for putting this together. Here is what I found:

## Issues

- An unauthenticated user gets 404 at https://narratives-pr-81.onrender.com/en/projects (after clicking in "Login")
- The profile picture and 'Manage My Narratives' button for authenticated users disappeared
- Sign Up page is not translated: https://narratives-pr-81.onrender.com/pt/signup
- Download my data is broken https://narratives-pr-81.onrender.com/download and [locale]/download
- Gallery link in the Portuguese Project panel leads to the English gallery (maybe automatic routing in my case)

- English-only title placeholder:
image

- English-only password emails:
image

## Comments

- Let's ask users for their preferred language when they register?
- Let's have a language field for each narrative, so we can filter the gallery by language?

**Comment by davidheyman** [comment]
@buccalon @martimpassos This is now ready for a final look and review. The big change is adding the additional translations around account management.

If we are getting close to your deadline, we can always merge this into master and then handle any outstanding changes there.

**Comment by martimpassos** [comment]
I'm getting a "unverified account" error/warning when trying to login in this deployment, same credentials work fine in production

**Comment by buccalon** [comment]
@davidheyman similar issue here. I tried to reset the password, but the action is not completed

image

**Comment by davidheyman** [comment]
@martimpassos It's a different database for the dev version so you'll need to create a new account.

**Comment by davidheyman** [comment]
@buccalon According to the database you did successfully reset your password. It's encrypted so I can't see what it is. Can you try again?

**Comment by martimpassos** [comment]
@davidheyman I created an account on the dev database. Inputting the correct password gives me `You could not be logged in. Please check your username and password.` An incorrect password leads to `Your account has not been verified. Please check your email for a verification link.`

**Comment by buccalon** [comment]
@davidheyman not working with my original dev account. I had to create another account by typing `/signup` in the URL, still, there was something wrong with the form's behavior. I can access the admin dashboard in with my original account on `/admin`

**Comment by buccalon** [comment]
@davidheyman 
- Account verification is broken (I had to verify @martimpassos manually)
- Gallery: Old URLs without language path return 404
- Loading animation English-only
- Search results English-only
image

**Comment by davidheyman** [comment]
These issues should now be resolved and this is ready for another test.

With the account verification, the accounts were being verified a second time on the translation redirect so I've disabled the failure message for double verification.

**Comment by buccalon** [comment]
Thanks, @davidheyman. This is what @martimpassos and I found:

## Issues

- Portuguese version has header menu in English
image

- Geographic layers are English-only and list two empty items
image

## Suggestions

We also felt that the language switch should be consistent across all imagineRio services (either header or footer).

## Comments

The only critical issue seems to be the English menu, so we agree in merging once it gets fixed.

**Pull request merged**